source/menu/events/add.ts

```diff
@@ -94,35 +85,76 @@ menu.choose('a', {
 	async choices(ctx) {
 		try {
 			const all = await findEvents(ctx);
-			return Object.fromEntries(all.map(event => [event.replaceAll('/', ';'), event]));
+			const alreadySelected = Object.keys(ctx.userconfig.mine.events);
+
+			return Object.fromEntries(all.map(element => {
+				if ('Id' in element) {
+					// Element is event
+					const event = element;
+
+					return alreadySelected.includes(event.Id) ? ['e:' + event.Id, '✅ ' + event.Name] : ['e:' + event.Id, '📅 ' + event.Name];
+				}
+
+				// Element is directory
+				const [dir, path] = element;
+
+				return dir.SubDirectories !== undefined || dir.Events !== undefined ? ['d:' + path.join(':'), '🗂️ ' + dir.Name] : ['dx:' + path.join(':'), '🚫 ' + dir.Name];
+			}));
 		} catch {
 			return {};
 		}
 	},
 	async do(ctx, key) {
-		const event = key.replaceAll(';', '/');
-		const isExisting = await allEventsExists(event);
-		const isAlreadyInCalendar = Object.keys(ctx.userconfig.mine.events)
-			.includes(event);
-
-		if (!isExisting) {
-			await ctx.answerCallbackQuery(`${event} existiert nicht!`);
+		if (key.startsWith('e:')) {
+			const eventId = key.slice(2);
+			const event = await allEventsExists(eventId);
+			const isAlreadyInCalendar = Object.keys(ctx.userconfig.mine.events).includes(eventId);
+
+			if (event === undefined) {
+				await ctx.answerCallbackQuery(`Event mit Id ${eventId} existiert nicht!`);
+				return true;
+			}
+
+			if (isAlreadyInCalendar) {
+				await ctx.answerCallbackQuery(`${event.Name} ist bereits in deinem Kalender!`);
+				return true;
+			}
+
+			ctx.userconfig.mine.events[eventId] = {name: event.Name};
+			await ctx.answerCallbackQuery(`${event.Name} wurde zu deinem Kalender hinzugefügt.`);
 			return true;
 		}
 
-		if (isAlreadyInCalendar) {
-			await ctx.answerCallbackQuery(`${event} ist bereits in deinem Kalender!`);
+		if (key.startsWith('d:')) {
+			ctx.session.eventPath = key.slice(2).split(':').map(Number);
+
 			return true;
 		}
 
-		ctx.userconfig.mine.events[event] = {};
-		await ctx.answerCallbackQuery(`${event} wurde zu deinem Kalender hinzugefügt.`);
-		return true;
+		await ctx.answerCallbackQuery('Dieses Verzeichnis ist leer und wird nur aus Gründen der Vollständigkeit dargestellt.');
```

Comment:
```suggestion
		await ctx.answerCallbackQuery('Dieses Verzeichnis ist leer');
```

Besonders auf mobilen Geräten sind die Texte schnell zu groß und auch schnell wieder weg.